src/qforte/abc/uccvqeabc.py

```diff
@@ -312,34 +312,29 @@ def report_iteration(self, x):
         self._k_counter += 1
 
         if self._k_counter == 1:
-            print(
-                "\n    k iteration         Energy               dE           Ngvec ev      Ngm ev*         ||g||"
-            )
-            print(
-                "--------------------------------------------------------------------------------------------------"
-            )
+            header = "\n    k iteration         Energy               dE"
+            if self._use_analytic_grad:
```

Comment:
I have a suggestion for future consideration. The decision to print gradient diagnostics should ideally depend on the type of optimizer being used, whether it is gradient-based or gradient-free. Even when numerical gradients are employed, gradient diagnostics should still be printed. However, `scipy.optimize.minimize` does not directly expose numerical gradients to the callback function. Therefore, the current decision to report gradient diagnostics for numerical gradients is based on `self._use_analytic_grad`.